api/Controllers/UserController.cs

```diff
@@ -1,76 +1,89 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using api.DTOs;
+using api.Models;
+using api.Service;
 using AutoMapper;
 using Microsoft.AspNetCore.Mvc;
-using YonderfulApi.DTOs;
-using YonderfulApi.Models;
-using YonderfulApi.Service;
 
-namespace YonderfulApi.Controllers
+namespace api.Controllers
 {
   [Route("api/[controller]")]
-    [ApiController]
-    public class UserController : ControllerBase
+  [ApiController]
+  public class UserController : ControllerBase
+  {
+    private readonly IUserService _userService;
+    private readonly IMapper _mapper;
+
+    public UserController(IUserService userService, IMapper mapper)
     {
-        private readonly IUserService _userService;
-        private readonly IMapper _mapper;
+      _userService = userService;
+      _mapper = mapper;
+    }
 
-        public UserController(IUserService userService, IMapper mapper)
-        {
-            _userService = userService;
-            _mapper = mapper;
-        }
+    [HttpGet("Id/{userId}")]
+    public async Task<IActionResult> GetUserById(int userId)
+    {
+      var user = await _userService.GetUserById(userId);
+      if (user == null)
+      {
+        return NotFound("No user with given id!");
+      }
+      return Ok(_mapper.Map<UserDto>(user));
```

Comment:
The UserDto now also contains the password.
I do not think that this should be sent to the FE when getting user info.
Maybe you should consider creating a new dto - UserDetailsDto - that does not contain the password